src/types.ts

```diff
@@ -0,0 +1,111 @@
+export interface DOMAPI {
+  createFragment: () => DocumentFragment;
+  createElement: (tagName: string) => HTMLElement;
+  createElementNS: (namespaceURI: string, qualifiedName: string) => Element;
+  createTextNode: (text: string) => Text;
+  createComment: (text: string) => Comment;
+  insertBefore: (parentNode: Node, newNode: Node, referenceNode: Node | null) => void;
+  removeChild: (node: Node, child: Node) => void;
+  appendChild: (node: Node, child: Node) => void;
+  parentNode: (node: Node) => Node | null;
+  nextSibling: (node: Node) => Node | null;
+  setTextContent: (node: Node, text: string) => void;
+}
+
+import {AttachData} from './helpers/attachto'
+import {VNodeStyle} from './modules/style'
+import {On} from './modules/eventlisteners'
+import {Attrs} from './modules/attributes'
+import {Classes} from './modules/class'
+import {Props} from './modules/props'
+import {Dataset} from './modules/dataset'
+import {Hero} from './modules/hero'
+
+export type Key = string | number;
+
+export type VNodes = Array<VNode | null>;
+
+export interface VNode {
+  nt: number; // matches DOM Spec for Node's nodeType
+  sel: string | undefined;
+  data: VNodeData;
+  children: VNodes | undefined;
+  elm: Node | undefined;
+  text: string | undefined;
+  key: Key | undefined;
+}
+
+export interface VElement extends VNode {
```

Comment:
this is just for convenience, and helps a lot to hint at the type system what kind of vnode is the code dealing with.